godtools/App/Features/Dashboard/Presentation/Tools/ToolsViewModel.swift

```diff
@@ -95,24 +97,17 @@ class ToolsViewModel: ObservableObject {
             }
             .store(in: &cancellables)
         
-        getUserFiltersUseCase.getUserFiltersPublisher()
-            .flatMap { userFilters in
+        $appLanguage.eraseToAnyPublisher()
+            .flatMap { appLanguage in
                 
-                return Publishers.CombineLatest(
-                    getToolFilterCategoriesUseCase.getCategoryFilterPublisher(with: userFilters.categoryFilterId),
-                    getToolFilterLanguagesUseCase.getLanguageFilterPublisher(from: userFilters.languageFilterId)
-                )
-                .eraseToAnyPublisher()
+                return self.getUserFiltersUseCase.getUserFiltersPublisher(translatedInAppLanguage: appLanguage)
+                    .eraseToAnyPublisher()
             }
-            .sink { categoryFilter, languageFilter in
-                
-                if let categoryFilter = categoryFilter {
-                    self.toolFilterCategorySelectionPublisher.send(categoryFilter)
-                }
-                
-                if let languageFilter = languageFilter {
-                    self.toolFilterLanguageSelectionPublisher.send(languageFilter)
-                }
+            .receive(on: DispatchQueue.main)
+            .sink { userFilters in
+              
+                self.toolFilterCategorySelectionPublisher.send(userFilters.categoryFilter)
```

Comment:
Capturing self within sink will cause a retain cycle. Will need to use capture list [weak self].